test/jdk/jdk/incubator/vector/TestMaskedMacroLogicVector.java

```diff
@@ -0,0 +1,493 @@
+        for (int i = 0; i < SPECIES.loopBound(r.length); i+=SPECIES.length()) {
+            VectorMask<Integer> vmask = VectorMask.fromArray(SPECIES, mask , i);
+            IntVector va = IntVector.fromArray(SPECIES, a, i);
+            IntVector vb = IntVector.fromArray(SPECIES, b, i);
+            IntVector vc = IntVector.fromArray(SPECIES, c, i);
+            vb.lanewise(VectorOperators.AND, va, vmask)
+            .lanewise(VectorOperators.XOR,
+                      vc.lanewise(VectorOperators.AND, va, vmask), vmask)
+           .intoArray(r, i);
+        }
+    }
+
+    static void verifyInt8(int[] r, int[] a, int[] b, int[] c, boolean [] mask) {
+        for (int i = 0; i < r.length; i++) {
+            int expected = intFunc8(a[i], b[i], c[i], mask[i]);
+            if (r[i] != expected) {
+                throw new AssertionError(String.format("testInt8: at #%d: r=%d, expected = %d = intFunc8(%d,%d,%d,%b)",
+                                                       i, r[i], expected, a[i], b[i], c[i], mask[i]));
+            }
+        }
+    }
+
+
+    // ===================================================== //
+
+    static long longFunc(long a, long b, long c) {
+        long v1 = (a & b) ^ (a & c) ^ (b & c);
+        long v2 = (~a & b) | (~b & c) | (~c & a);
+        return v1 & v2;
+    }
+
+    static void testLong(VectorSpecies<Long> SPECIES, long[] r, long[] a, long[] b, long[] c) {
+        for (int i = 0; i < SPECIES.loopBound(r.length); i += SPECIES.length()) {
+            LongVector va = LongVector.fromArray(SPECIES, a, i);
+            LongVector vb = LongVector.fromArray(SPECIES, b, i);
+            LongVector vc = LongVector.fromArray(SPECIES, c, i);
+
+            va.lanewise(VectorOperators.AND, vb)
+            .lanewise(VectorOperators.XOR, va.lanewise(VectorOperators.AND, vc))
+            .lanewise(VectorOperators.XOR, vb.lanewise(VectorOperators.AND, vc))
+            .lanewise(VectorOperators.AND,
+                       va.lanewise(VectorOperators.NOT).lanewise(VectorOperators.AND, vb)
+                      .lanewise(VectorOperators.OR, vb.lanewise(VectorOperators.NOT).lanewise(VectorOperators.AND, vc))
+                      .lanewise(VectorOperators.OR, vc.lanewise(VectorOperators.NOT).lanewise(VectorOperators.AND, va)))
+            .intoArray(r, i);
+        }
+    }
+
+    static void verifyLong(long[] r, long[] a, long[] b, long[] c) {
+        for (int i = 0; i < r.length; i++) {
+            long expected = longFunc(a[i], b[i], c[i]);
+            if (r[i] != expected) {
+                throw new AssertionError(
+                        String.format("testLong: at #%d: r=%d, expected = %d = longFunc(%d,%d,%d)",
+                                      i, r[i], expected, a[i], b[i], c[i]));
+            }
+        }
+    }
+
+    // ===================================================== //
+
+    private static final Random R = new Random(1024);
```

Comment:
It would be better to use Utils.getRandomInstance() from jdk.test.lib.Utils to ensure that the seed is printed for reproducibility.